broker/broker-routemanager/src/main/java/org/codice/ddf/broker/routemanager/DynamicRouteDeployer.java

```diff
@@ -13,134 +13,102 @@
+            LOGGER.debug(e.getMessage(), e);
+        } catch (IOException e) {
+            LOGGER.warn("IO Exception when accessing {}. See debug log for stack trace.",
+                    file.getName());
+            LOGGER.debug(e.getMessage(), e);
+        } catch (Exception e) {
+            LOGGER.warn("Failed to read route definition. See debug log for stack trace.");
+            LOGGER.debug(e.getMessage(), e);
+        }
     }
 
-    private void processFile(Path path, boolean added) {
-        if (added) {
-            try (InputStream is = new FileInputStream(path.toFile())) {
-                List<RouteDefinition> routeDefinitions;
-
-                LOGGER.info("Loading route path: {}", path.getFileName());
-
-                routeDefinitions = camelContext.loadRoutesDefinition(is)
-                        .getRoutes();
-
-                camelContext.addRouteDefinitions(routeDefinitions);
-                processedFiles.put(path, routeDefinitions);
-                camelContext.startAllRoutes();
+    @Override
+    public void update(File file) throws Exception {
+        uninstall(file);
+        install(file);
+    }
 
-            } catch (FileNotFoundException e) {
-                LOGGER.warn("File {} not found. See debug log for stack trace.",
-                        path.getFileName());
-                LOGGER.debug(e.getMessage(), e);
-            } catch (IOException e) {
-                LOGGER.warn("IO Exception when accessing {}. See debug log for stack trace.",
-                        path.getFileName());
-                LOGGER.debug(e.getMessage(), e);
-            } catch (Exception e) {
-                LOGGER.warn("Failed to read route definition. See debug log for stack trace.");
-                LOGGER.debug(e.getMessage(), e);
-            }
-        } else {
-            LOGGER.info("Removing route path: {}", path.getFileName());
+    @Override
+    public void uninstall(File file) throws Exception {
+        try (InputStream is = new FileInputStream(file)) {
+            LOGGER.info("Removing route path: {}", file.getName());
             try {
-                camelContext.removeRouteDefinitions(processedFiles.get(path));
+                camelContext.removeRouteDefinitions(getProcessedFiles().get(file));
             } catch (Exception e) {
                 LOGGER.warn("Failed to remove routes definition. See debug log for stack trace.");
                 LOGGER.debug(e.getMessage(), e);
             }
-            processedFiles.remove(path);
+            processedFiles.remove(file);
         }
     }
 
-    public void destroy() throws Exception {
-        camelContext.stop();
+    @Override
+    public boolean canHandle(File file) {
+        return file.getName()
+                .endsWith(".xml");
     }
+
+    public Map<File, List<RouteDefinition>> getProcessedFiles() {
+        if (processedFiles == null) {
```

Comment:
I was going to override that method in my unit test for the `update` method but then I ended up just using mocks. It seemed a little cleaner to use the method, but It certainly is simpler to just create a `new HashMap<>()`. Fixed.